libraries/AP_Scripting/applets/set_home_position.lua

```diff
@@ -0,0 +1,127 @@
+-- Script to set new home position with the RC
+-- Sets the arming, preset or dynamic home position after 2 seconds that the switch is in the chosen position
+-- Author: Marco Robustini, Alessandro Apostoli
+
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+-- setup param block, reserving 2 params beginning with HSW_
+local PARAM_TABLE_KEY = 74
+local PARAM_TABLE_PREFIX = 'HSW_'
+assert(param:add_table(PARAM_TABLE_KEY, "HSW_", 4), 'could not add param table')
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+    assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+    return Parameter(PARAM_TABLE_PREFIX .. name)
+end
+
+--[[
+  // @Param: HSW_ENABLE
+  // @DisplayName: Home switch enable
+  // @Description: Home switch enable
+--]]
+HSW_ENABLE = bind_add_param('ENABLE', 1, 0)
+
+--[[
+  // @Param: HSW_PRESET_LAT
+  // @DisplayName: Preset Home location latitude
+  // @Description: Preset Home location latitude
+--]]
+HSW_PRESET_LAT = bind_add_param('PRESET_LAT', 2, 0) -- example value 44.123456
+
+--[[
+  // @Param: HSW_PRESET_LNG
+  // @DisplayName: Preset Home location longitude
+  // @Description: Preset Home location longitude
+--]]
+HSW_PRESET_LNG = bind_add_param('PRESET_LNG', 3, 0) -- example value 11.123456
+
+--[[
+  // @Param: HSW_PRESET_ALT
+  // @DisplayName: Preset Home location altitude
+  // @Description: Preset Home location altitude
+  // @Units cm
+--]]
+HSW_PRESET_ALT = bind_add_param('PRESET_ALT', 4, 0) -- example value 1000 = 10m
+
+local rc_option = 300
+local sw_channel = rc:find_channel_for_option(rc_option)
```

Comment:
```suggestion
local sw_channel = assert(rc:find_channel_for_option(rc_option), "RCx_OPTION 300 not configured")
```
Can be `nil` if the switch is not setup.

Reply:
I will update it with all the changes, thank you @IamPete1!